### PR TITLE
Update Automata for the Enso Lexer

### PR DESCRIPTION
**Pull request by iamrecursion**
### Pull Request Description
This PR fixes the automata library for the use-case of the flexer.

### Important Notes
Bumps `enso-lazy-reader` to `0.1.1` and bumps `enso-automata` to `0.1.4`.

### Checklist
Please include the following checklist in your PR:

- [x] The documentation has been updated if necessary.
- [x] All code conforms to the [Rust](https://github.com/enso-org/enso/blob/main/docs/style-guide/rust.md) style guide.
- [x] All code has been tested where possible.

### CODE REVIEW

File: src/automata/src/pattern.rs [hunk 1]
```diff
@@ -46,17 +45,17 @@ impl Pattern {
...
     pub fn min() -> Self {
-        Pattern::symbol(&Symbol::new_named(0,"MIN"))
+        Pattern::symbol(&Symbol::null())
     }
 
     /// A pattern that triggers on the minimum value.
     pub fn max() -> Self {
-        Pattern::symbol(&Symbol::new_named(u64::max_value(),"MAX"))
+        Pattern::symbol(&Symbol::eof())
     }
```
Comment (farmaazon) [inline src/automata/src/pattern.rs:59]: I would suggest `::from(SymbolIndex::min_value())`, `::from(SymbolIndex::max_value())`, or making functions `min` and `max` in Symbol. The fact that null is minimum, and EOF is maximum seems incidental.

File: src/automata/src/symbol.rs [hunk 2]
```diff
@@ -1,6 +1,15 @@
 //! Defines a Symbol that is operated on by the finite automata.
 
 use crate::prelude::*;
+use std::cmp::Ordering;
```
Comment (farmaazon) [inline src/automata/src/symbol.rs:4]:
```suggestion

use std::cmp::Ordering;
```

File: src/automata/src/symbol.rs [hunk 3]
```diff
@@ -9,22 +18,24 @@ use crate::prelude::*;
 // ==============
 
 /// An input symbol to a finite automaton.
-#[derive(Clone,Debug,PartialEq,Eq,PartialOrd,Ord,Hash)]
+#[derive(Clone,Debug)]
```
Comment (farmaazon) [inline src/automata/src/symbol.rs:21]: Why we stopped to derive Eq and Hash? Usually it's in a good taste to implement the common traits if possible https://rust-lang.github.io/api-guidelines/interoperability.html

**Review by farmaazon — APPROVED** [review]

File: src/automata/src/symbol.rs (hunk 3 above)
Comment (iamrecursion) [reply src/automata/src/symbol.rs:21]: They're implemented by hand below! The name field shouldn't contribute to the definition of equality, ordering, or hashing.

**Pull request merged**